control-plane/pkg/reconciler/broker/broker.go

```diff
@@ -572,8 +572,17 @@ func storeConfigMapAsStatusAnnotation(broker *eventing.Broker, cm *corev1.Config
 		broker.Status.Annotations = make(map[string]string, len(cm.Data))
 	}
 
+	keysToStore := map[string]bool{
+		kafka.DefaultTopicNumPartitionConfigMapKey:      true,
+		kafka.DefaultTopicReplicationFactorConfigMapKey: true,
+		kafka.BootstrapServersConfigMapKey:              true,
+		security.AuthSecretNameKey:                      true,
```

Comment:
@basit9958 I don't think that we need to keep this key - the config map doesn't normally have this key (as far as I can tell). See here: https://github.com/knative-extensions/eventing-kafka-broker/blob/8a79f31bea81dcb2d94a0bb3524f080fd66ea9c3/control-plane/config/eventing-kafka-broker/100-broker/100-kafka-broker-configmap.yaml

cc @matzew @creydr in case I am missing something
```suggestion
```